met-api/src/met_api/services/engagement_slug_service.py

```diff
@@ -0,0 +1,79 @@
+"""Service for engagement slug management."""
+from met_api.models.engagement_slug import EngagementSlug as EngagementSlugModel
+from met_api.models.engagement import Engagement as EngagementModel
+from met_api.constants.engagement_status import Status
+from met_api.services.slug_generation_service import SlugGenerationService
+
+
+class EngagementSlugService:
+    """Service for engagement slug management."""
+
+    @classmethod
+    def get_engagement_slug(cls, slug: str) -> EngagementSlugModel:
+        """Get an engagement slug by slug."""
+        engagement_slug = EngagementSlugModel.find_by_slug(slug)
+        if not engagement_slug:
+            raise ValueError(f'No engagement slug found for {slug}')
+        return engagement_slug
+
+    @classmethod
+    def create_engagement_slug(cls, engagement_id: int) -> EngagementSlugModel:
+        """Create an engagement slug."""
+        existing_slug = EngagementSlugModel.find_by_engagement_id(engagement_id)
+        if existing_slug:
+            raise ValueError(f'Engagement slug already exists for engagement {engagement_id}')
+
```

Comment:
may be dont throw error ;just dont generate
slug generation shouldnt affect the actual functionality from where it gets invoked